gossipd/routing.c

```diff
@@ -77,7 +77,6 @@ struct node *new_node(struct routing_state *rstate,
 	n->out = tal_arr(n, struct node_connection *, 0);
 	n->alias = NULL;
 	n->node_announcement = NULL;
-	n->last_timestamp = 0;
```

Comment:
We are returning the channels when asked for them with `getchannels`. If we issue a `getchannels` before receiving the `channel_announcement` and the `channel_update` valgrind will yell at us.

We ourselves will never produce a timestamp that is 0, the above messages refer to duplicate updates that we either sent ourselves and get reflected back from our peers or updates that have travelled two different paths. We can change this to be a signed int64 and use -1 to indicate a missing update if you'd like.

Reply:
Yes, I thought about -1.  Let's do that...